### PR TITLE
Better reset support

### PR DESCRIPTION
**Pull request by rowleya**
Updated tests for SpiNNakerManchester/sPyNNaker#669

### CODE REVIEW

**Comment by rowleya** [comment]
See also: http://apollo.cs.man.ac.uk:8080/blue/organizations/jenkins/sPyNNaker8%20Integration%20Tests/detail/better_reset_support/6/pipeline

**Comment by rowleya** [comment]
A number of tests were set to skip due to the bugs the related PRs fix, and failed when not allowed to skip.  This has now been fixed as well, and seemed to be the tests rather than the code that were causing the issues.

File: spynnaker8/spinnaker.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def is_a_pynn_random(self, thing):
         return isinstance(thing, RandomDistribution)
 
     def get_pynn_NumpyRNG(self):
-        return NumpyRNG()
+        return NumpyRNG
```
Comment (alan-stokes) [inline spynnaker8/spinnaker.py]: this seems unrelated

Comment (rowleya) [reply spynnaker8/spinnaker.py]: This is essential to the changes in sPyNNaker.  This allows us to re-run the random with the original seed, which is needed if connectivity is to be regenerated.

Comment (alan-stokes) [reply spynnaker8/spinnaker.py]: k.
